src/lib/templateengine.js

```diff
@@ -1371,6 +1376,14 @@ function TemplateEngine( undefined ) {
 
   this.create_pages = function(page, path, flavour, type) {
     var creator = thisTemplateEngine.design.getCreator(page.nodeName);
+    
+    thisTemplateEngine.callbacks[ path + '_' ] = {
+      exitingPageChange: [],// called when the current page is left
+      beforePageChange: [], // called as soon as a page change is known
+      duringPageChange: [], // called when the page is theoretical visible, i.e. "display:none" is removed - CSS calculations shoud work now
+      afterPageChange: []   // called together with the global event when the transition is finished
```

Comment:
Do we really need before, during and afterPageChange? What would happen if strip them down to only
 enteringPage, leavingPage (and maybe beforeEnteringPage) (IMHO those names would fit better)?